app/models/comment.rb

```diff
@@ -0,0 +1,31 @@
+class Comment < ActiveRecord::Base
+  belongs_to :article
+  attr_accessible :content, :email, :subject
+  validates :subject, presence: true
+  validates :email, presence: true
+  validates :article_id, presence: true
+  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+  after_commit :_flush_all_cache
+
+  def cached_article
+    Rails.cache.fetch(["article_comment", self.id]) do
```

Comment:
@phendych two issues:
- you can just use Article's cache by using Article.cached_find(self.article_id) because Comment has article_id
- how are you going to invalidate this cache?